### ISSUE TITLE
How to build files from different source files, located in different packages in sourcedir src/main/java?

### ISSUE THREAD

**Issue by pirancaci** [asciidoctor/asciidoctor-gradle-plugin#532]

**Comment by ysb33r** [comment]
Assuming your have `src/main/java/foo/bar/a.adoc` and `src/main/java/alice/bob/b.adoc`, then your starting point is
```
asciidoctor {
  sourceDir 'src/main/java'
}
asciidoctorPdf {
  sourceDir 'src/main/java'
}
```
which can build `build/docs/asciidoc/foo/bar/a.html`, `build/docs/asciidoctor/alice/bob/b.html`, `build/docs/asciidoctorPdf/foo/bar/a.pdf` etc.

I suspect that is not quite what you want though. I suspect you want something akin to `alice.html`, `alice.pdf`, `foo.html`, `foo.pdf` which will include all of the asciidoct file below `src/main/java/alice` and `src/main/java/foo` respectively?

If you can explain the last part in more detail, I might have a solution for you.

**Comment by pirancaci** [comment]
> Assuming your have `src/main/java/foo/bar/a.adoc` and `src/main/java/alice/bob/b.adoc`, then your starting point is
> 
> ```
> asciidoctor {
>   sourceDir 'src/main/java'
> }
> asciidoctorPdf {
>   sourceDir 'src/main/java'
> }
> ```
> 
> which can build `build/docs/asciidoc/foo/bar/a.html`, `build/docs/asciidoctor/alice/bob/b.html`, `build/docs/asciidoctorPdf/foo/bar/a.pdf` etc.
> 
> I suspect that is not quite what you want though. I suspect you want something akin to `alice.html`, `alice.pdf`, `foo.html`, `foo.pdf` which will include all of the asciidoct file below `src/main/java/alice` and `src/main/java/foo` respectively?
> 
> If you can explain the last part in more detail, I might have a solution for you.

Hello, thank you so much for your comment. You really helped me to solve my problem. I got what I wanted. I was missing the line 'sourceDir 'src/main/java'. 
I have now another problem, which has to do with the include directive.
Usually I use this directive which works fine when the asciidoc files are located in src/docs/asciidoc: include::{sourcedir}/example/example.java[] with sourcedir: ../../src/main/java

Now that I have asciidoc files located in src/main/java/package/docs the include directive is not working and gives me error message. I would be very thankful if you could help me with this issue as well. 

This is the error message: 

Multiple markers at this line
- Validation failed:include::.. references not existing file:C:\Users\pcaci\Desktop\GenerateDoc\src\main\java\move\docs\{sourcedir}\move\Move.java
- include file not found: C:/Users/pcaci/Desktop/GenerateDoc/src/main/java/main/java/move/Move.java
[
<img width="691" alt="Untitled" src="https://user-images.githubusercontent.com/35587443/75897772-62aeef80-5e39-11ea-89e3-0dd623f9edf1.png">
](url)

**Comment by ysb33r** [comment]
Is `{sourcedir}` defined as an attribute?

**Comment by pirancaci** [comment]
yes it is defined as an attribute on the top of asciidoc file and in the asciidoctor task in the build.gradle file. 
<img width="768" alt="1" src="https://user-images.githubusercontent.com/35587443/75902596-8c1f4980-5e40-11ea-8ca0-7b1d7b4863d7.png">

<img width="649" alt="1" src="https://user-images.githubusercontent.com/35587443/75903015-3ac38a00-5e41-11ea-91a3-a33f2631e3db.png">

**Comment by ysb33r** [comment]
This output `:C:\Users\pcaci\Desktop\GenerateDoc\src\main\java\move\docs{sourcedir}\move\Move.java` looks like `{sourcedir}` has the wrong value.

**Comment by pirancaci** [comment]
> This output `:C:\Users\pcaci\Desktop\GenerateDoc\src\main\java\move\docs{sourcedir}\move\Move.java` looks like `{sourcedir}` has the wrong value.

yes I understand there is a problem, but I don't know why. Since I don't have experience with sourceDir other then ../../main/java as in the gradle plugin examples. 
I discovered another issue, which is the images. They don't appear on the converted html file, only on the pdf file. I couldn't find a solution for that as well. I assume there is something wrong witht the imagedir, but couldn't find out why. Maybe you have any idea? Thanks in advance

**Comment by pirancaci** [comment]
Hello,

Please can you help me with this?

I don't know why the image is not appearing in html and the include
directive is giving me error . Maybe you have a solution Please let me know
:D I tried a lot, I don't what to do anymore.

Thank in advance

**Comment by ysb33r** [comment]
There are a number of issues in your build.

1. You should use the `asciidoctorPdf` task for PDF generation.
2. You should use the `resources { include 'path/to/images/**' }` block to point to your image location.

Also temporarily do 
```
asciidoctor {
  doFirst {
    println "******* ${sourceSets.main.java.srcDirs[0]}"
  }
}
```

This might help you diagnose why your source directory is not correct.

**Comment by mojavelinux** [comment]
It would be best to continue these types of discussions in the project chat. I recommend closing this issue and picking it back up there if there's still more to discuss.
